### ISSUE TITLE
MIT Interop - Support default_ccache_name FILE method

### ISSUE THREAD

**Issue by quanah** [heimdal/heimdal#355]
Starting with 1.11, MIT supports things like:

```
[libdefaults]
default_ccache_name = FILE:/var/krb5/security/creds/krb5cc_%{uid}
```

If this configuration is used with Heimdal, problems ensue.  For example, "klist" will segfault.

**Comment by saper** [comment]
This works for me as of heimdal 7.5.0

````
[libdefaults]
    default_realm = EXAMPLE.ORG
    default_cc_name = FILE:/tmp/krb5cc_zz_%{uid}
````

and then

````
% /usr/local/bin/kinit saper@EXAMPLE.ORG
saper@EXAMPLE.ORG's Password: 
saper@poniatowski:~ % /usr/local/bin/klist 
Credentials cache: FILE:/tmp/krb5cc_zz_169
        Principal: saper@EXAMPLE.ORG

  Issued                Expires               Principal
Feb 22 21:35:56 2018  Feb 23 21:35:56 2018  krbtgt/EXAMPLE.ORG@EXAMPLE.ORG
````

Unfortunately, an attempt to use `DIR:` crashes.

**Comment by quanah** [comment]
default_cc_name != default_ccache_name

**Comment by lhoward** [comment]
It shouldn't crash, but default_ccache_name is not recognised by Heimdal. Just following up that this issue is about adding support for this, not `DIR:` – looking at the code, it appears that cache enumeration in unsupported for that.

**Comment by lhoward** [comment]
Fixed in 6d7b0bfd. Also added support for expansion of {euid} and {username} tokens in the interest of greater MIT compatibility.
